testonly/hammer/hammer.go

```diff
@@ -309,11 +306,23 @@ func newHammerState(ctx context.Context, cfg *MapConfig) (*hammerState, error) {
 		cfg.OperationDeadline = 60 * time.Second
 	}
 
+	var prevContents testonly.VersionedMapContents
+	var smrs smrStash
+	validReadOps := validReadOps{
+		vc:              vc,
+		extraSize:       cfg.ExtraSize,
+		chooseLeafCount: func(prng *rand.Rand) int { return pickIntInRange(cfg.MinLeaves, cfg.MaxLeaves, prng) },
```

Comment:
Is this random int selector over-engineered? Both `hammerState` and `validReadOps` have a copy of this function (with the implementation that appears to be the same). Also, there is a type definition for it. But in the end, there is only one implementation that calls `pickIntInRange`. Maybe remove the type/method/function, and call `pickIntInRange` directly? For example, you could throw away the `extraSize` field in favor of a full `cfg`, and then use `cfg.Min/MaxLeaves` in `validReadOps`.

Reply:
Part of the motivation behind this PR is to encapsulate better. Anything with a reference to a `MapConfig` effectively has access to the whole world. That is too big of a handbag and makes it hard to reason about what code is up to.

I could pass in a min/max instead of this function, but this way kinda naturally happened as a way of making the intent of the read operations as clear as possible. Happy to pass in min/max and copy the impl of pickIntInRange into validReadOps if that is clearer.

Reply:
Yeah, I think it would be clearer.

In the future maybe it would make sense to create sub-configs of `cfg`, and copy the corresponding sub-config into `validReadOps`.